lib/postprocessing/watermarkProcessor.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const { getDimensions } = require('./assetProperties');
+const { ImageOperations } = require('./imageOperations');
+
+class WatermarkProcessor {
+    constructor(instructions) {
+
+        this.renditionWidth = instructions.width || this.getRenditionDimensions(instructions.path).width || 0;
+        this.renditionHeight = instructions.height || this.getRenditionDimensions(instructions.path).height || 0;
+        this.watermarkWidthPercent = instructions.watermark.widthPercent || 100;
+        this.watermarkPath = instructions.watermark.path;
+        this.getWatermarkDimensions(this.watermarkPath);
+    }
+
+    getRenditionDimensions (path) {
```

Comment:
why does this have to be its own function?
I would just call `getDimensions` directly on line 20:
```js
renditionDimensions = getDimensions(instructions.path);
this.renditionWidth = renditionDimensions.width || 0;
this.renditionHeight = renditionDimensions.height || 0;
```

Reply:
`getRenditionDimensions()` may not even need to be called at all - It seemed cleaner and easier to read

Reply:
Ack with @jdelbick .

It might make sense if you have `getRenditionDimensions()` and `getWatermarkDimensions()` without the `path` arguments. And the paths are known by the object (they kind of are already). Then it would follow the object encapsulation. Right now the path is passed around again and again and nothing guarantees it is the same.